File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/iin/IinDetailsRequest.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.iin;

import com.onlinepayments.sdk.client.android.model.PaymentContext;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * POJO that contains the request for IIN lookup.
 */
public class IinDetailsRequest implements Serializable {

	private static final long serialVersionUID = 8401271765455867950L;

	@SerializedName("bin")
	private String ccPartial;

	private PaymentContext paymentContext;

	/**
	 @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public IinDetailsRequest(String ccPartial) {
		this.ccPartial = ccPartial;
	}

	/**
	 @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public IinDetailsRequest(String ccPartial, PaymentContext paymentContext) {
		this.ccPartial = ccPartial;
		this.paymentContext = paymentContext;
	}

	public String getCcPartial() {
		return ccPartial;
	}

	public PaymentContext getPaymentContext () {
		return paymentContext;
	}

}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/iin/IinDetailsRequest.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.iin

import com.google.gson.annotations.SerializedName
import com.onlinepayments.sdk.client.android.model.PaymentContext
import java.io.Serializable

/**
 * Data class that contains the request for IIN lookup.
 */
data class IinDetailsRequest(
    @SerializedName("bin") val ccPartial: String?,
    val paymentContext: PaymentContext? = null
) : Serializable {

    companion object {
        @Suppress("Unused")
        private val serialVersionUID = 8401271765455867950L
    }
}
